students/Roy_Tate/lesson04/mailroom2.py

```diff
@@ -0,0 +1,184 @@
+# author/student: Roy Tate (githubtater)
+
+from collections import Counter
+
+import sys
+import os
+import os.path
+
+# Create the donor_dict and manually add a few entries
+donor_dict = {}
+donor_dict['Bill'] = [1100, 2100, 3100]
+donor_dict['Mark'] = [1000, 300000.23, 2000]
+donor_dict['Jeff'] = [50, 450.355]
+donor_dict['Roy'] = [4500, 7500, 11221, 30232, 323]
+
+# This is the standard template to be used when sending thank you emails. Ideally this would be saved
+# externally as a template file.
+email_str_ty = '''\n
+        From:     A Worthwhile Cause
+        To:       {0}
+        Subject:  Thank you
+
+        Dear {0},
+
+        We want to thank you for your recent donation of ${1:.2f}. This was
+        very generous.
+
+        Sincerely,
+
+        The Boss
+        '''
+
+# Email template used for generating thank you emails to all donors.
+email_str_annual = '''\n
+        From:    A Charity Thankful For Your Kindness
+        To:      {0}
+        Subject:  This Year's Challenge!
+
+        Dear {0},
+
+        First, we would like to thank you for your continued generosity throughout the 
+        years. Without contributions like yours, the good things that we are able to do 
+        simply would not be possible. 
+
+        Your contributions to date have totaled ${1:.2f}. 
+
+        This years challenge is to see if you can donate more than your current total
+        in one year. This would effectively double your current donations to the 
+        organization!  Remember, it is for a good cause!
+
+        Sincerely,
+
+        The good guys at the best organization'''
+
+
+# Prints the list of keys from donor_dict (in this case, the donor names)
+def print_list():
+    print('{:*^50}'.format('DONOR LIST'))
+    for donor in donor_dict.keys():
+        print(donor)
+
+
+# Helper function to add donations to appropriate donor's account
+def add_donation(name, new_donation):
+    if name in donor_dict.keys():
+        donor_dict[name].append(new_donation)
+    else:
+        donor_dict[name] = [new_donation]
+
+
+# Send formatted email to donor
+def print_email(email_str, donor_name, donation_amount):
```

Comment:
The `email_str` parameter is not being used in your function, but the `email_str_ty` variable in your function block is referencing the variable defined in the global namespace so the function still works